engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -535,11 +535,19 @@ class PeriodicProcessService(
         case PeriodicProcessDeploymentStatus.Failed =>
           Some(ScheduledExecutionStatus.Failed)
         case PeriodicProcessDeploymentStatus.RetryingDeploy =>
-          Some(ScheduledExecutionStatus.DeploymentWillBeRetried)
+          None
         case PeriodicProcessDeploymentStatus.FailedOnDeploy =>
-          Some(ScheduledExecutionStatus.DeploymentFailed)
+          None
       }
-    } yield (status, dateFinished)
+      dateCreated = instantAtSystemDefaultZone(entity.createdAt)
```

Comment:
the existence of the `instantAtSystemDefaultZone` def forces me to ask the following questions:
1. why do we operate on LocalDateTime, not on Instant in the domain? are we able to improve it?
2. shouldn't we have the `ZonedDateTime` in the API responses rather than Instant? (aka, "consume utc dates, return dates with zone" principle)?

Reply:
Current state overall:
- we operate on LocalDateTime here, because that is how they are stored in the periodic db
- this db is the main problem - there is no info about timezone for periodics at all
- even searching for deployment to run is performed on local times. So theoretically it would fail, if system timezone would change between scheduling and running (or some other hypothetical instance of Nu is running on different timezone - it uses Clock.systemDefaultZone)

This specific case:
- I may be wrong, but AFAIK all, or nearly all, APIs operate on Instant's
- instantAtSystemDefaultZone is a conversion caused by the problems related by storing LocalDateTime's in the periodic db
- API returns dates in two ways: 
  - as Instant's, then it is formatted by FE
  - as generic params, as plain text - then is it formatted by BE in systemDefaultZone. The BE implementation is prepared to accept timezone in request, but it is not done ATM. (There is comment 'TODO: pass client timezone from FE'  for similar case - pdf generation

Reply:
Refactored, using ZonedDateTime, and conversions on repo level.

Reply:
discussed and reverted because of some problems detected by tests, left for later